doc/src/build/package-versioning.md

```diff
@@ -0,0 +1,30 @@
+
+Note that there is also a notion of versioning present in the [manifest](./move/manifest.md) files. You can observe it both in the [package section](./move/manifest.md#package-section) and in the [dependencies section](./move/manifest.md#dependencies-section). For example:
+```move
+[package]
+name = "some_pkg"
+version = "1.0.0"
+
+[dependencies]
+another_pkg = { git = "https://github.com/another_pkg/another_pkg.git" , version = "2.0.0"}
+```
+These fields in the manifest file, however, at this point are only used for user-level documentation purposes and are ignored by the build or publish/upgrade commands. Another way of looking at it is that if a developer publishes a package with a certain package version in the manifest file and then modifies and re-publishes the same package with a different version (using publish command rather than upgrade command), these two packages will be considered different packages, rather than on-chain versions of the same package, and recognized as such by the tooling - for example, none of these packages could be used as a [dependency override](./dependency-overrides.md) to stand in for the other one.
+
+## Version management
+
+In order to control the evolution of a published package, a developer can implement a [custom upgrade policy](./custom-upgrade-policy.md) but Sui also comes with a set of built-in upgrade policies. For an overview of how an upgrade policy should be applied, see [here](./custom-upgrade-policy.md#upgrade-overview).
+
+There are four built-in package upgrade policies, listed here in the order of strictness:
+- immutable - this policy is a bit of a misnomer as having it applied means that the package cannot be upgraded at all
+- dependency-only - the only thing that can be modified in an upgraded version of this package are its dependencies
+- additive - new functionality can be added to the package (e.g., new public functions or structs) but none of the existing functionality can be changed (e.g., the code in existing public functions cannot change)
+- compatible - the most relaxed policy where in addition to everything allowed by more restrictive policies, an upgraded version of the package can change the code of existing public functions, signatures of existing functions and definitions of existing types cannot change
```

Comment:
Some more things that can happen in a Compatible upgrade:

- Ability constraints on generic type parameters in function signatures can be removed, 
- `private`, `public(friend)`, and `entry` functions can be changed, removed, or made `public`.

This leads me to a question for @tzakian -- do we support making a private function public in additive mode?  It seems like we should, technically speaking.

Finally, a table seems like the most suitable way to visualize this information, perhaps followed by a description of what each kind of change is. @ronny-mysten / @randall-Mysten, how feasible would that be?

Reply:
Updated but left in the list form for now.